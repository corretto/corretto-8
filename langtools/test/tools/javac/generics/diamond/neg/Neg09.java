/*
 * @test /nodynamiccopyright/
 * @bug 7020044
 *
 * @summary  Check that diamond is not allowed with anonymous inner class expressions
 * @author Maurizio Cimadamore
 * @compile/fail/ref=Neg09.out Neg09.java -XDrawDiagnostics
 *
 */

class Neg09 {
    class Member<X> {}

    static class Nested<X> {}

    void testSimple() {
        Member<?> m1 = new Member<>() {};
        Nested<?> m2 = new Nested<>() {};
    }

    void testQualified() {
        Member<?> m1 = this.new Member<>() {};
        Nested<?> m2 = new Neg09.Nested<>() {};
    }
}
