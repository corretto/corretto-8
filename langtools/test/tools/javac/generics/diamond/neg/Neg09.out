Neg09.java:17:34: compiler.err.cant.apply.diamond.1: Neg09.Member<X>, (compiler.misc.diamond.and.anon.class: Neg09.Member<X>)
Neg09.java:18:34: compiler.err.cant.apply.diamond.1: Neg09.Nested<X>, (compiler.misc.diamond.and.anon.class: Neg09.Nested<X>)
Neg09.java:22:39: compiler.err.cant.apply.diamond.1: Neg09.Member<X>, (compiler.misc.diamond.and.anon.class: Neg09.Member<X>)
Neg09.java:23:40: compiler.err.cant.apply.diamond.1: Neg09.Nested<X>, (compiler.misc.diamond.and.anon.class: Neg09.Nested<X>)
4 errors
